script/acceptancetest

```diff
@@ -7,97 +7,15 @@ source `dirname $0`/stackenv
-# acceptance/openstack/baremetal/v1
-
-acceptance/openstack/blockstorage
-acceptance/openstack/blockstorage/extensions
-
-# snapshots_test.go:16: Unable to create a blockstorage client: CinderEndpoint is required
-# acceptance/openstack/blockstorage/noauth
-
-# snapshots_test.go:21: Unable to retrieve snapshots: Resource not found
-# acceptance/openstack/blockstorage/v1
-# acceptance/openstack/blockstorage/v2
-
-acceptance/openstack/blockstorage/v3
-
-# No suitable endpoint could be found in the service catalog.
-# acceptance/openstack/clustering/v1
-
-acceptance/openstack/compute/v2
-
-# No suitable endpoint could be found in the service catalog.
-# acceptance/openstack/containerinfra/v1
-
-acceptance/openstack/container/v1
-
-# Unable to create a DB client: No suitable endpoint could be found in the service catalog.
-# acceptance/openstack/db/v1
-
-acceptance/openstack/dns/v2
-acceptance/openstack/identity/v2
-acceptance/openstack/identity/v3
-acceptance/openstack/imageservice/v2
-acceptance/openstack/keymanager/v1
-acceptance/openstack/loadbalancer/v2
-
-# No suitable endpoint could be found in the service catalog.
-# acceptance/openstack/messaging/v2
-
-acceptance/openstack/networking/v2
-acceptance/openstack/networking/v2/extensions
-acceptance/openstack/networking/v2/extensions/agents
-acceptance/openstack/networking/v2/extensions/dns
-
-# Resource not found
-# acceptance/openstack/networking/v2/extensions/fwaas
-# acceptance/openstack/networking/v2/extensions/fwaas_v2
-acceptance/openstack/networking/v2/extensions/layer3
-
-# Unable to create: Resource not found
-# acceptance/openstack/networking/v2/extensions/lbaas
-
-# Resource not found
-# acceptance/openstack/networking/v2/extensions/lbaas_v2
-
-acceptance/openstack/networking/v2/extensions/mtu
-acceptance/openstack/networking/v2/extensions/networkipavailabilities
-acceptance/openstack/networking/v2/extensions/portsbinding
-acceptance/openstack/networking/v2/extensions/qos/policies
-acceptance/openstack/networking/v2/extensions/qos/rules
-acceptance/openstack/networking/v2/extensions/qos/ruletypes
-acceptance/openstack/networking/v2/extensions/quotas
-acceptance/openstack/networking/v2/extensions/rbacpolicies
-acceptance/openstack/networking/v2/extensions/subnetpools
-acceptance/openstack/networking/v2/extensions/trunks
-acceptance/openstack/networking/v2/extensions/vlantransparent
-
-# Unable to create: Resource not found
-# acceptance/openstack/networking/v2/extensions/vpnaas
-
-acceptance/openstack/objectstorage/v1
-acceptance/openstack/orchestration/v1
-acceptance/openstack/placement/v1
-acceptance/openstack/sharedfilesystems/v2
-acceptance/openstack/sharedfilesystems/v2/messages
-
-# No suitable endpoint could be found in the service catalog
-# acceptance/openstack/workflow/v2
-)
+if [[ -z $ACCEPTANCE_TESTS ]]; then
+    echo "No acceptance tests to run"
+    exit 0
+fi
 
-for acceptance_test in ${ACCEPTANCE_TESTS[@]}; do
+for acceptance_test in "${ACCEPTANCE_TESTS[@]}"; do
```

Comment:
Remove the quotes, it causes the loop to fail:
```
2021-11-09 16:07:40.477103 | ubuntu-focal | + ./script/acceptancetest:main:16          :   for acceptance_test in "${ACCEPTANCE_TESTS[@]}"
2021-11-09 16:07:40.484427 | ubuntu-focal | + ./script/acceptancetest:main:17          :   go test -v -timeout 60m -tags 'fixtures acceptance' ./acceptance/openstack/blockstorage acceptance/openstack/blockstorage/extensions acceptance/openstack/blockstorage/noauth acceptance/openstack/blockstorage/v1 acceptance/openstack/blockstorage/v2 acceptance/openstack/blockstorage/v3 acceptance/openstack/imageservice/v2 acceptance/openstack/objectstorage/v1 acceptance/openstack/sharedfilesystems/v2 acceptance/openstack/sharedfilesystems/v2/messages
2021-11-09 16:07:40.538842 | ubuntu-focal | package acceptance/openstack/blockstorage/extensions is not in GOROOT (/usr/local/go/src/acceptance/openstack/blockstorage/extensions)
2021-11-09 16:07:40.539032 | ubuntu-focal | package acceptance/openstack/blockstorage/noauth is not in GOROOT (/usr/local/go/src/acceptance/openstack/blockstorage/noauth)
2021-11-09 16:07:40.539145 | ubuntu-focal | package acceptance/openstack/blockstorage/v1 is not in GOROOT (/usr/local/go/src/acceptance/openstack/blockstorage/v1)
2021-11-09 16:07:40.539269 | ubuntu-focal | package acceptance/openstack/blockstorage/v2 is not in GOROOT (/usr/local/go/src/acceptance/openstack/blockstorage/v2)
2021-11-09 16:07:40.539386 | ubuntu-focal | package acceptance/openstack/blockstorage/v3 is not in GOROOT (/usr/local/go/src/acceptance/openstack/blockstorage/v3)
2021-11-09 16:07:40.539510 | ubuntu-focal | package acceptance/openstack/imageservice/v2 is not in GOROOT (/usr/local/go/src/acceptance/openstack/imageservice/v2)
2021-11-09 16:07:40.539621 | ubuntu-focal | package acceptance/openstack/objectstorage/v1 is not in GOROOT (/usr/local/go/src/acceptance/openstack/objectstorage/v1)
2021-11-09 16:07:40.539738 | ubuntu-focal | package acceptance/openstack/sharedfilesystems/v2 is not in GOROOT (/usr/local/go/src/acceptance/openstack/sharedfilesystems/v2)
2021-11-09 16:07:40.540412 | ubuntu-focal | package acceptance/openstack/sharedfilesystems/v2/messages is not in GOROOT (/usr/local/go/src/acceptance/openstack/sharedfilesystems/v2/messages)
```

https://logs.openlabtesting.org/logs/52/2252/6ceef7690d7fef7fddfb83756c8446d2c63a5139/check/gophercloud-acceptance-test-storage/df72cf6/job-output.txt.gz

Reply:
@mandre this patch causes too much troubles for now, I want to do this, but in a next step, and make it clean.
For now, I would like to restore a good shape for CI and this PR is really too big.. I don't want to make it bigger.

Thanks

Reply:
Fair enough, let's merge this PR.
I created https://github.com/gophercloud/gophercloud/issues/2267 to keep track of the skipped tests.